bitnami/jupyterhub/templates/_helpers.tpl

```diff
@@ -0,0 +1,158 @@
+{{/*
+Return the proper singleuser image name (to be set in the hub.configuration part). We cannot use common.images.image because of the tag
+{{ include "jupyterhub.hubconfiguration.imageEntry" ( dict "imageRoot" .Values.path.to.the.image "global" $) }}
+*/}}
+{{- define "jupyterhub.hubconfiguration.imageEntry" -}}
+{{- $registryName := .imageRoot.registry -}}
+{{- $repositoryName := .imageRoot.repository -}}
+{{- if .global }}
+    {{- if .global.imageRegistry }}
+     {{- $registryName = .global.imageRegistry -}}
+    {{- end -}}
+{{- end -}}
+{{- if $registryName }}
+{{- printf "%s/%s" $registryName $repositoryName -}}
+{{- else -}}
+{{- printf "%s" $repositoryName -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the proper hub image name
+*/}}
+{{- define "jupyterhub.proxy.image" -}}
+{{ include "common.images.image" (dict "imageRoot" .Values.proxy.image "global" .Values.global) }}
+{{- end -}}
+
+{{/*
+Return the proper hub image name
+*/}}
+{{- define "jupyterhub.auxiliary.image" -}}
+{{ include "common.images.image" (dict "imageRoot" .Values.auxiliaryImage "global" .Values.global) }}
+{{- end -}}
+
+{{/*
+Return PostgreSQL postgres user password
+*/}}
+{{- define "jupyterhub.proxyToken" -}}
+{{- if .Values.proxy.secretToken -}}
+    {{- .Values.proxy.secretToken -}}
+{{- else -}}
+    {{- randAlphaNum 32 -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the proper Docker Image Registry Secret Names
+*/}}
+{{- define "jupyterhub.imagePullSecrets" -}}
+{{- include "common.images.pullSecrets" (dict "images" (list .Values.hub.image .Values.proxy.image .Values.auxiliaryImage) "global" .Values.global) -}}
+{{- end -}}
+
+{{/*
+Create the name of the service account to use
+*/}}
+{{- define "jupyterhub.hubServiceAccountName" -}}
+{{- if .Values.hub.serviceAccount.create -}}
+    {{ default (printf "%s-hub" (include "common.names.fullname" .)) .Values.hub.serviceAccount.name }}
+{{- else -}}
+    {{ default "default" .Values.hub.serviceAccount.name }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Create the name of the service account to use
+*/}}
+{{- define "jupyterhub.singleuserServiceAccountName" -}}
+{{- if .Values.singleuser.serviceAccount.create -}}
+    {{ default (printf "%s-singleuser" (include "common.names.fullname" .)) .Values.singleuser.serviceAccount.name }}
+{{- else -}}
+    {{ default "default" .Values.singleuser.serviceAccount.name }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Create a default fully qualified postgresql name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+*/}}
+{{- define "jupyterhub.postgresql.fullname" -}}
+{{- $name := default "postgresql" .Values.postgresql.nameOverride -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Return  the proper Storage Class
+*/}}
+{{- define "jupyterhub.storageClass" -}}
+{{- include "common.storage.class" (dict "persistence" .Values.singleuser.persistence "global" .Values.global) -}}
+{{- end -}}
```

Comment:
Why not using directly the "common.storage.class" macro in the corresponding place?

Reply:
Done